returns/transducers/tfilter.py

```diff
@@ -0,0 +1,41 @@
+from typing import Callable, TypeVar
+
+_ValueType = TypeVar('_ValueType')
+_AccValueType = TypeVar('_AccValueType')
+
+
+def tfilter(
+    predicate: Callable[[_ValueType], bool],
+) -> Callable[
+    [Callable[[_AccValueType, _ValueType], _AccValueType]],
+    Callable[[_AccValueType, _ValueType], _AccValueType],
+]:
+    """
+    :py:func:`filter <filter>` implementation on a transducer form.
+
+    .. code:: python
+
+      >>> from typing import List
+      >>> from returns.transducers import tfilter, reduce
+
+      >>> def is_even(number: int) -> bool:
+      ...     return number % 2 == 0
+
+      >>> def append(collection: List[int], item: int) -> List[int]:
+      ...     collection.append(item)
+      ...     return collection
+
+      >>> my_list = [0, 1, 2, 3, 4, 5, 6]
+      >>> xform = tfilter(is_even)(append)
+      >>> assert reduce(xform, my_list, []) == [0, 2, 4, 6]
+
+    """
+    def reducer(
```

Comment:
We tend to use `decorator` / `factory` naming when defining nested decorators

Reply:
Yep, but I think using that pattern might be weird:
```python
def tmap(function):
    def reducer(step):
        def map_(acc, value):
            ...
        return map_
    return reducer

def tmap(function):
    def decorator(step):
        def factory(acc, value):
            ...
        return map_
    return reducer
```

In fact transducers is not intended to be use as a decorator and I think `tmap -> reducer -> map_` is more meaningful! 🤔